sample-apps/go-sample-app/collection/random_metrics.go

```diff
@@ -0,0 +1,173 @@
+package collection
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"time"
+
+	"go.opentelemetry.io/otel/metric/global"
+	"go.opentelemetry.io/otel/metric/instrument"
+	"go.opentelemetry.io/otel/metric/instrument/asyncint64"
+	"go.opentelemetry.io/otel/metric/instrument/syncint64"
+)
+
+var (
+	meter               = global.MeterProvider().Meter("OTLP_METRIC_SAMPLE_APP")
+	threadsActive int64 = 0
+	threadsBool         = true
+)
+
+// randomMetricCollector contains all the random based metric instruments.
+type randomMetricCollector struct {
+	timeAlive     syncint64.Counter
+	cpuUsage      asyncint64.Gauge
+	heapSize      asyncint64.UpDownCounter
+	threadsActive syncint64.UpDownCounter
+}
+
+// NewRandomMetricCollector returns a new type struct that holds and registers the 4 random based metric instruments used in the Go-Sample-App;
+// HeapSize, ThreadsActive, TimeAlive, CpuUsage
+func NewRandomMetricCollector() randomMetricCollector {
```

Comment:
This should accept a `MeterProvider` and create its own `Meter` from it.

Reply:
I updated this as well as in RequestBasedMetricCollector